test/framework/clusterctl/client.go

```diff
@@ -0,0 +1,155 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterctl
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	clusterctlclient "sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+	clusterctllog "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl/logger"
+)
+
+// Provide E2E friendly wrappers for the clusterctl client library.
+
+// InitInput is the input for Init.
+type InitInput struct {
+	LogPath                 string
+	ClusterctlConfigPath    string
+	KubeconfigPath          string
+	CoreProvider            string
+	BootstrapProviders      []string
+	ControlPlaneProviders   []string
+	InfrastructureProviders []string
+}
+
+// Init calls clusterctl init with the list of providers defined in the local repository
+func Init(ctx context.Context, input InitInput) {
+	By(fmt.Sprintf("clusterctl init --core %s --bootstrap %s --control-plane %s --infrastructure %s",
+		input.CoreProvider,
```

Comment:
This question is related to the current kubeconfig file management, that is something that I would like to improve in a separate PR/after some considerations because it is potentially a breaking change.
TL;DR; Currently kubeconfig path is a random file under tmp folder, which makes it kind of unusable. Instead, IMO the kubeconfig path should be configurable, so we can store it in the artifact folder and used it for debugging.